exonum-java-binding-qa-service/src/main/java/com/exonum/binding/qaservice/QaServiceImpl.java

```diff
@@ -166,6 +171,17 @@ public HashCode submitUnknownTx() {
     });
   }
 
+  /**
+   * Increments the initial counter so the number of times this method was invoked is stored in it.
+   */
+  @Override
+  public void afterCommit(BlockCommittedEvent event) {
+    long seed = 1L;
+    HashCode counterId = Hashing.sha256()
+        .hashString(INITIAL_COUNTER_NAME, StandardCharsets.UTF_8);
+    submitIncrementCounter(seed, counterId);
```

Comment:
Won't it work forever? `Initialize -> new block -> after commit -> new block -> after commit -> new block`. Or mb I missunderstood something?

Reply:
it'll spam the network and create the new transaction for each block.

Reply:
Is new block created on every transaction?

Reply:
Exonum produces blocks with parameterized constant frequency. If no transaction received then empty block generates. So, therefore, this method will create the transaction for each block 🤔

Reply:
Sorry, I don't see why is that a bad thing. We allow services to submit transactions in `afterCommit` method. If they want to submit a transaction on every block commit - why shouldn't they?

Reply:
Every blockchain would work forever, won't it? :)

Reply:
@dip56 :arrow_up: :warning: 

@bullet-tooth Are you saying that the system will never be idle, i.e., there will always be a transaction, triggering the next block as soon as `min_propose_timeout`? 

@MakarovS is there a setting that defines how many transactions there must be to use `min_propose_timeout`?

Reply:
Yes, as you mentioned it's `propose_timeout_threshold` - https://github.com/exonum/exonum-java-binding/pull/571/files/99e7f9e644726b119e992b242e25e609d4b91278..31ec5c477ea24654538c922f91fef77d875a73ee#diff-53ec26b2abc32dea308e36d7e4efdc4aR79
Also there is `txs_block_limit` - maximum number of transactions per block